src/App.tsx

```diff
@@ -1,26 +1,68 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
-
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42)
-  .map(n => `Item ${n}`);
+import { Pagination } from './components/Pagination';
 
 export const App: React.FC = () => {
+  const [perPage, setPerPage] = useState(5);
+  const [currentPage, setCurrentPage] = useState(1);
+
+  const getArrOfItemsForCurrPage = (
+    sum: number,
+    sumOfPosPerPage: number,
+    currPage: number,
+  ) => {
+    const startPosition = currPage === 1
+      ? 1
+      : sumOfPosPerPage * currPage - sumOfPosPerPage + 1;
+
+    const finalPosition = sumOfPosPerPage * currPage > sum
+      ? sum
+      : sumOfPosPerPage * currPage;
+
+    const arrayOfItems = getNumbers(startPosition, finalPosition);
+
+    return arrayOfItems;
+  };
+
+  const handleAmountOfItems = (
+    event: React.ChangeEvent<HTMLSelectElement>,
+  ): void => {
+    const { value } = event.target;
+
+    setCurrentPage(1);
+    setPerPage(+value);
+  };
+
+  const handlePage = (event: React.MouseEvent<HTMLAnchorElement>): void => {
+    const { dataset } = event.currentTarget;
+
+    setCurrentPage(Number(dataset.value));
+  };
+
+  const currentItems = getArrOfItemsForCurrPage(42, perPage, currentPage);
+
+  const startPos = currentItems[0];
+
+  const endPos = currentItems[(currentItems.length - 1)];
```

Comment:
you can calculate it based on data you have in this component

Reply:
Thanks for the advice, I hope I did everything right.